spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/InterceptorAutoConfiguration.java

```diff
@@ -0,0 +1,62 @@
+package org.axonframework.springboot.autoconfig;
+
+import org.axonframework.commandhandling.CommandBus;
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.commandhandling.gateway.CommandGateway;
+import org.axonframework.config.ConfigurerModule;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.MessageDispatchInterceptor;
+import org.axonframework.messaging.MessageHandlerInterceptor;
+import org.axonframework.queryhandling.QueryBus;
+import org.axonframework.queryhandling.QueryMessage;
+import org.axonframework.spring.config.SpringConfigurer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+
+import java.util.List;
+
+@AutoConfiguration
+@ConditionalOnClass(SpringConfigurer.class)
+@AutoConfigureAfter({
+        AxonAutoConfiguration.class,
+        JpaAutoConfiguration.class,
+        JpaEventStoreAutoConfiguration.class,
+        NoOpTransactionAutoConfiguration.class,
+        TransactionAutoConfiguration.class
+})
+public class InterceptorAutoConfiguration {
+
+    @ConditionalOnClass(MessageDispatchInterceptor.class)
+    @Autowired
+    public void commandDispatchInterceptorConfigurer(CommandGateway commandGateway, List<MessageDispatchInterceptor<? super CommandMessage<?>>> interceptors) {
+        interceptors.forEach(i -> commandGateway.registerDispatchInterceptor(i));
+    }
+
+    @ConditionalOnClass(MessageHandlerInterceptor.class)
```

Comment:
In the first approach I've tried to do it with a `ConfigurerModule` but I wasn't able to get to the needed components. AFAIK the `Configurer` lacks a get() method for registered components (I can only register new components). 

Using @Autowired looks a bit weird at first but with this I get the primary CommandGateway etc. after they have been initialized.

For events I also needed to use the `EventProcessingConfigurer` because there are many event handlers and there was a `registerDefaultHandlerInterceptor` method.

Reply:
Sorry, I am not being specific enough, I think.
With the `ConfigurerModule`, you get a `Configurer` as the given parameter.
On the `Configurer`, you can use the `Configurer#onInitialize(Consumer<Configuration>)` method.
It is there that on a given phase, you can retrieve infra components, like the `CommandGateway`, and register interceptors.

In short pseudo-code, something like this:

```java
ConfigurerModule configurerModule = configurer -> configurer.onInitialize(
        config -> config.commandGateway().registerDispatchInterceptor({my-interceptor-here})
);
```

Then again, I don't really mind which route we take for this! Just wanted to understand why you took this direction. Feel free to leave it at `@Autowired` methods :+1: